src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/mobile/put_file_request.cc

```diff
@@ -311,9 +311,12 @@ void PutFileRequest::SendOnPutFileNotification() {
   message[strings::msg_params][strings::app_id] = connection_key();
   message[strings::msg_params][strings::sync_file_name] = sync_file_name_;
   message[strings::msg_params][strings::offset] = offset_;
-  if (0 == offset_) {
-    message[strings::msg_params][strings::file_size] =
-        (*message_)[strings::msg_params][strings::length];
+  if ((*message_)[strings::msg_params].keyExists(strings::offset)) {
+    message[strings::msg_params][strings::offset] = offset_;
+    if (0 == offset_ && (*message_)[strings::msg_params].keyExists(strings::length)) {
+      message[strings::msg_params][strings::file_size] =
+          (*message_)[strings::msg_params][strings::length];
```

Comment:
Wrong case, this should be set to `length_` if `keyExists(strings::length)`, `file_size` should not be set if `length` is sent by the app. This is because if `length` is provided or `offset` is nonzero, it means that this is a partial data chunk, which means that you cannot determine the length of the file.

Reply:
@jacobkeeler please take a look b4691c7b3190119afa10a6bc77f1c8315b82a743